tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,622 @@
+}
+
+func DeployClusterWithValues(t *testing.T, options, customValues string) {
+	t.Logf("Deploying a cluster with %s options using the %s values", options, customValues)
+
+	helmValues := map[string]string{}
+	if options == "default" {
+		helmValues = map[string]string{
+			"reaper.ingress.host": "repair.localhost",
+		}
+	}
+	if options == "minio" {
+		serviceName := MinioServiceName(t)
+		helmValues = map[string]string{
+			"medusa.storage_properties.host": fmt.Sprintf("%s.minio.svc.cluster.local", strings.ReplaceAll(serviceName, "'", "")),
+		}
+	}
+	deployCluster(t, customValues, helmValues)
+}
+
+func DeployClusterWithValuesAndHeapSettings(t *testing.T, options, cassandraHeap, stargateHeap, customValues string) {
+	t.Logf("I can deploy a cluster with %s options, %s Cassandra Heap and %s Stargate heap using the %s values", options, cassandraHeap, stargateHeap, customValues)
+
+	splitOptions := strings.Split(options, "-")
+	medusaEnabled := "true"
+	reaperEnabled := "true"
+	monitoringEnabled := "true"
+
+	if Find(splitOptions, "nomedusa") {
+		medusaEnabled = "false"
+	}
+
+	if Find(splitOptions, "noreaper") {
+		reaperEnabled = "false"
+	}
+
+	if Find(splitOptions, "nomonitoring") {
+		monitoringEnabled = "false"
+	}
+
+	newGenSize, _ := strconv.Atoi(strings.ReplaceAll(strings.ReplaceAll(cassandraHeap, "M", ""), "G", ""))
+	helmValues := map[string]string{
+		"cassandra.heap.size":           cassandraHeap,
+		"cassandra.heap.newGenSize":     strconv.Itoa(newGenSize/2) + "M",
+		"stargate.heapMB":               strings.ReplaceAll(stargateHeap, "M", ""),
+		"medusa.enabled":                medusaEnabled,
+		"reaper.enabled":                reaperEnabled,
+		"reaper-operator.enabled":       reaperEnabled,
+		"kube-prometheus-stack.enabled": monitoringEnabled,
+	}
+	deployCluster(t, customValues, helmValues)
+}
+
+func CheckK8ssandraNamespaceWasCreated(t *testing.T) {
+	t.Logf("I can see the %s namespace in the list of namespaces", namespace)
+
+	kubectlOptions := k8s.NewKubectlOptions("", "", "default")
+	_, err := k8s.GetNamespaceE(t, kubectlOptions, namespace)
+	g(t).Expect(err).To(gomega.BeNil())
+}
+
+func CheckSecretIsPresent(t *testing.T, secret string) {
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	_, err := k8s.GetSecretE(t, kubectlOptions, secret)
+	g(t).Expect(err).To(gomega.BeNil())
+}
+
+func CheckNamespaceIsAbsent(t *testing.T) {
+	g(t).Eventually(func() bool {
+		return namespaceIsAbsentOrTerminating(t, namespace)
+	}, RETRY_TIMEOUT, RETRY_INTERVAL).Should(gomega.BeTrue())
+}
+
+func namespaceIsAbsentOrTerminating(t *testing.T, ns string) bool {
+	kubectlOptions := k8s.NewKubectlOptions("", "", "default")
+	namespaceObject, _ := k8s.GetNamespaceE(t, kubectlOptions, namespace)
+	if namespaceObject == nil {
+		// namespace was deleted
+		return true
+	}
+
+	if namespaceObject.Status.Phase == v1.NamespaceTerminating {
+		// namespace is terminating, which is good enough
```

Comment:
Namespace deleted and terminating are not equal. Namespace can stay in terminating phase forever if there's finalizers blocking the deletion. That would be a bug in uninstall.

Reply:
The problem then is the time scenarios will take as I've seen namespace deletions take several minutes to complete.
Do we want to test that the namespace was fully deleted this in all scenarios?
Maybe we could create a test scenario that would be specific to this in order to shorten the overall duration?

Reply:
The duration for namespace deletion is increased due to the time it takes for the CassandraDatacenter deletion (see #570 and #571 for details).

@adejanovski after uninstalling the chart and before deleting the namespace, wait for the termination of the CassandraDatacenter. See https://github.com/jsanda/medusa-operator/blob/e2e-in-place-restore/test/framework/framework.go#L123 for an example.

Reply:
Done: https://github.com/k8ssandra/k8ssandra/pull/544/files#diff-06af14725d9bc4e9eef72fdfd0d188ffbc8a3e657d3999dc6d1346a1eb37b9fdR53